addon/models/resource.js

```diff
@@ -388,12 +388,12 @@ const RelatedProxyUtil = Ember.Object.extend({
     @param {Ember.ObjectProxy|Ember.ArrayProxy} proxyFactory
     @return {PromiseProxy} proxy
   */
-  createProxy: function (resource, proxyFactory) {
+  createProxy: function (resource, type, proxyFactory) {
     const relation = this.get('relationship');
     const url = this.proxyUrl(resource, relation);
-    const service = resource.container.lookup('service:' + pluralize(relation));
+    const service = resource.container.lookup('service:' + pluralize(type));
     let promise = this.promiseFromCache(resource, relation, service);
-    promise = promise || service.findRelated(relation, url);
+    promise = promise || service.findRelated(type, url);
```

Comment:
The API returned a URL for the relation which is still used for the `url` when calling `service.findRelated`; however, it seems that the service you want to use should be configurable based on either the (default) `resource` name or an optional `type` name